src/ert/run_models/run_arguments.py

```diff
@@ -0,0 +1,75 @@
+from dataclasses import dataclass, field
+from typing import List, Optional
+
+
+@dataclass
+class SingleTestRunArguments:
+    simulation_mode: str = field(init=False, default="Single test run")
+
+    active_realizations: List[bool]
+    current_case: str
+    target_case: str
+    start_iteration: Optional[int] = 0
+    num_iterations: Optional[int] = 1
+    prev_successful_realizations: Optional[int] = 0
+
+    iter_num: Optional[int] = 0
+    item_number: Optional[int] = 0
+
+
+@dataclass
+class EnsembleExperimentRunArguments:
+    simulation_mode: str = field(init=False, default="Ensemble experiment")
+
+    active_realizations: List[bool]
+    current_case: str
+    target_case: str
+    iter_num: int
+    item_number: Optional[int] = 0
+    start_iteration: Optional[int] = 0
+    num_iterations: Optional[int] = 1
+    prev_successful_realizations: Optional[int] = 0
+
+
+@dataclass
+class ESRunArguments:
+    simulation_mode: str = field(init=False, default="Ensemble smoother")
+    analysis_module: str = field(init=False, default="STD_ENKF")
+
+    active_realizations: List[bool]
+    current_case: str
+    target_case: str
+    start_iteration: Optional[int] = 0
+    num_iterations: Optional[int] = 1
+    prev_successful_realizations: Optional[int] = 0
+
+
+@dataclass
+class ESMDARunArguments:
+    simulation_mode: str = field(init=False, default="Multiple data assimilation")
+    analysis_module: str = field(init=False, default="STD_ENKF")
+
+    active_realizations: List[bool]
+    target_case: str
+    weights: str
```

Comment:
Instead of using `str` here, you can have the type be `Sequence[float]` and have a Pydantic validator convert from `str`. Move the code from `MultipleDataAssimilation.parseWeights` here as a validator.

This makes it so that the correctness of `weights` is an invariant, rather something that the user code has to remember to call.

Reply:
In this PR, I want to make the smallest change possible that replaces `Dict[str, Any]` with dataclasses for simulation arguments.